proxy/src/main/java/com/velocitypowered/proxy/command/SuggestionsProvider.java

```
@@ -0,0 +1,367 @@
+        // We intentionally don't catch all unchecked exceptions
+        child.parse(reader, context);
+        if (reader.canRead() && reader.peek() != CommandDispatcher.ARGUMENT_SEPARATOR_CHAR) {
+          continue;
+        }
+      } catch (final CommandSyntaxException e) {
+        continue;
+      }
+      if (reader.canRead(2)) { // separator + string
+        reader.skip(); // separator
+        final ParseResults<S> parse = this.parseHints(child, reader, context);
+        if (potentials == null) {
+          potentials = new ArrayList<>(1);
+        }
+        potentials.add(parse);
+      }
+    }
+    if (potentials != null) {
+      if (potentials.size() > 1) {
+        potentials.sort((a, b) -> {
+          if (!a.getReader().canRead() && b.getReader().canRead()) {
+            return -1;
+          }
+          if (a.getReader().canRead() && !b.getReader().canRead()) {
+            return 1;
+          }
+          return 0;
+        });
+      }
+      return potentials.get(0);
+    }
+    return new ParseResults<>(contextSoFar, originalReader, Collections.emptyMap());
+  }
+
+  /**
+   * Returns a future that is completed with the result of merging the {@link Suggestions}
+   * the given futures complete with. The results of the futures that complete exceptionally
+   * are ignored.
+   *
+   * @param fullInput the command input
+   * @param futures the futures that complete with the suggestions
+   * @return the future that completes with the merged suggestions
+   */
+  @SafeVarargs
+  private CompletableFuture<Suggestions> merge(
+          final String fullInput, final CompletableFuture<Suggestions>... futures) {
+    // https://github.com/Mojang/brigadier/pull/81
+    return CompletableFuture.allOf(futures).handle((unused, throwable) -> {
+      final List<Suggestions> suggestions = new ArrayList<>(futures.length);
+      for (final CompletableFuture<Suggestions> future : futures) {
+        if (!future.isCompletedExceptionally()) {
```

Would it be possible to at least log exceptionally-completed futures? Swallowing exceptions in whole makes for difficult debugging.